### ISSUE TITLE
Improving the license workflow acceptance documentation

### ISSUE THREAD

**Issue by vmbrasseur** [spdx/license-list-XML#1052]
Notes from working on #993.

The doc in question: https://github.com/spdx/license-list-XML/blob/master/DOCS/new-license-workflow.md#if-license-is-accepted

Reviewing 'If license is accepted'

- [x] Link `SPDX Online tools` in 7.i.
    * https://spdxtools.sourceauditor.com/
- [ ] 7.i assumes that the license exists in the License Requests list
    * In the case of #767, the CC-BY-NC-ND-3.0-IGO license isn't in the list.
- [x] 7.i.a could use a link to a "wrapped text version" example.
- [x] 7.i "edit XML" requires a login
    * Mention this in the doc
    * Only signin option is 'Login with GitHub'
- [x] 7.i "edit XML" is a button at the bottom of the license text (License Requests > click license > scroll to the bottom
    * This is NOT the 'License XML Editor' link from the top of the page
- [x] 7.i.b should include directions for where `listVersionAdded=` should be added, and perhaps an example same.
- [ ] 7.i.c says "standard license header" as though it's common knowledge. It's not. An example is required.
    * I don't know where to find this or to what it refers.
- [ ] 7.i.d has a broken link to an Overview page.
- [x] 7.i.f should link to a license to use as an example. It also should explain "the bullet or list tags" better. Where would these tags be found?
- [x] 7.ii says to run the tests locally, but the changes are happening using the online tool. There has to be a "copy your changes locally, then…" step in there.
    * Also, at no point was the user directed to clone the repo locally. Probably should ask them to do that? But if they're doing all the work in the online tool, maybe just drop the local test step?

**Comment by vmbrasseur** [comment]
I was going to assign this to myself, but it appears I lack the access to assign things. If someone with access could please do the needful…?

**Comment by swinslow** [comment]
wonderful, thanks @vmbrasseur for pulling these notes together! assigned now.

**Comment by swinslow** [comment]
Example text from an error:

```
steve@clariel: license-list-XML> ./test-one-license BSD-3-Clause
gpg --verify --no-default-keyring --keyring ./goneall.gpg licenseListPublisher-2.1.22.jar.asc
gpg: assuming signed data in 'licenseListPublisher-2.1.22.jar'
gpg: Signature made Tue Oct 15 21:58:59 2019 EDT
gpg:                using RSA key 8CCBE63DE91BFF1C
gpg: Good signature from "Gary O'Neall (Key for signing jar files) <gary@sourceauditor.com>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: EA76 0AF7 CBD0 E4F9 013F  06FB 8CCB E63D E91B FF1C
java -jar -DLocalFsfFreeJson=false -DlistedLicenseSchema="schema/ListedLicense.xsd" licenseListPublisher-2.1.22.jar LicenseRDFAGenerator src '.tmp' 1.0 2000-01-01 test/simpleTestForGenerator expected-warnings
ERROR StatusLogger No log4j2 configuration file found. Using default configuration: logging only errors to the console. Set system property 'log4j2.debug' to show Log4j2 internal initialization logging.
Processing License List.
Processing Exceptions
Writing table of contents
The following warning(s) were identified:
	Test for license BSD-3-Clause failed: Test for license ID BSD-3-Clause failed due to difference found Normal text of license does not match starting at line #11 column #26 "by" when comparing to template text " be used to endorse or promote products derived from this software without specific prior written pe..."
Completed processing licenses
make: *** [validate-canonical-match] Error 64
```

**Comment by vmbrasseur** [comment]
# Notes from the call on 2020-07-02

* Steve walking through what he does to add a license
    * From the XML adding step
* Have a license we've decided to accept
    * 2 parts to add
        * XML file: markup of the license
        * Test file: plain text of the license for testing
    * Doesn't normally use the GH editor, uses VSCode
* 0BSD as an example
    * XML file
    * test/SimpleTestForGenerator
* CI/CD
    * Takes markup
    * Looks for validity
    * Parse the text
    * Compare against the test text version
* Steve usually starts with the test file
    * Then it'll be there first
* XML file…
    * GSoC intern built an online XML editor
        * VMB: This is the one that's documented to use
        * SW: Have been complications w/whitespace, etc
    * SW finds a prior one that's similar then copies it, edits appropriately
        * If a new license, do the same thing but just to get the correct header/footer start tags
* Some potential Gotchas
    * Name of XML file must be same as the identifier used for the license
        * Same for test file
        * And licenseId in the XML header
        * licenseId is immutable; we will never change it after this
            * May deprecate, but will never change
    * isOsiApproved is true/false
    * name is the title that displays
    * In newer licenses, have liseVersionAdded in header
        * CI/CD doesn't check for this
        * We add to older licenses as we bother editing them
        * Please include it
* Other fields
    * crossRefs
        * Point people to…
            * URLs where people can find the license text
                * Canonical text for a license with a steward like Apache
            * URLs where people used to be able to find the license text
                * A link to a place where someone used the license, for instance
        * We never remove or modift these
            * Keep if the link breaks
            * Having these URLs as cross references is helpful for cases where people used the URL in the past
        * Will add them whenever it makes sense
    * notes
        * What it says on the tin
        * Might have a note to explain what the license is meant to be doing, how it differs from something else it looks like, etc.
    * Formatting
        * Wrap paragraphs in p tags
        * If there's a list in the text, wrap in list tags (bullet or number)
            * See BSD-3-Clause-Open-MPI for an example
            * Bullet itself matches as "bullet", not as the bullet character, so easier to pass CI/CD
    * alt match=
        * Regex for variations on a text but all still the same license
        * Example: BSD-3-Clause-Open-MPI
        * Want all the variations to match as the same license; just minor wording differences
        * Often start w/plain text then add alt matches later as they come up
* Testing
    * How to test w/o submitting PR & hoping?
    * Run local tests
    * Publisher tool written in Java
        * Java required to run tests
    * `make validate-canonical-match`
        * Actual command run by the CI/CD system
        * Confirms have latest version of pub tools
        * Will test every license in the list
    * `./test-one-license` command
        * Give it the ID of the license
        * Renames all other files, runs test on that license, un-renames the files after tests
    * Pub tool ensures don't have identical licenses
        * Tests every license against every other one to ensure against duplicates
        * So helpful to run full tests rather than just `test-one-license`
        * Are some cases expect licenses to have same text
            * For instance, GFDL licenses have same text but different identifiers
            * For these, can add to `expected-warnings` file

**Comment by swinslow** [comment]
Updated checkbox list above to mark the ones that were fixed in #1070. I'll submit a PR to finish off the others.

**Comment by swinslow** [comment]
Added #1081 which will finalize this.
